### PR TITLE
Introduce HttpController interface for http routers

### PR DESCRIPTION
**Pull request by junlarsen**
By requiring controllers to declare their own routes, we ensure consistency between tests and the actual server

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -872,13 +872,9 @@ async function main() {
...
         })
         .use(bodyParser.json({limit: ceProps('bodyParserLimit', maxUploadSize)}))
-        .get('/api/siteTemplates', siteTemplateController.getSiteTemplates.bind(siteTemplateController))
-        .get('/source/:source/list', sourceController.listEntries.bind(sourceController))
-        .get('/source/:source/load/:language/:filename', sourceController.loadEntry.bind(sourceController))
-        .get(
-            '/api/asm/:arch/:opcode',
-            assemblyDocumentationController.getOpcodeDocumentation.bind(assemblyDocumentationController),
-        )
+        .use('/', siteTemplateController.createRouter())
```
Comment (mattgodbolt) [inline app.ts:875]: is the `/` necessary? I am never quite sure but it seems like (e.g.) the above bodyParser doesn't need a route?

**Review by mattgodbolt — APPROVED** [review]
you know far more than me but the `/` root binding is a bit surprising

File: app.ts (hunk 1 above)
Comment (junlarsen) [reply app.ts:875]: Maybe not, I can do a test without

Comment (junlarsen) [reply app.ts]: you're right! :star:

**Pull request merged**